### DISCUSSION TITLE
Does TSDF reconstruction using VoxelBlockGrid support spatial carving?

### DISCUSSION THREAD

**Discussion by martinakos** [isl-org/Open3D#6286] [Q&A]
I've made a scene reconstruction system based on the examples/python/t_reconstruction_system/integrate.py example. 
This is a TSDF reconstruction using the VoxelBlockGrid data structure. Its use is described in the Documentation http://www.open3d.org/docs/release/tutorial/t_reconstruction_system/integration.html

My system keeps repeating these two main steps:

-Updates the VoxelBlockGrid using the the compute_unique_block_coordinates()  and integrate() functions. 
-It periodicallly extracts a pointcloud from the VoxelBlockGrid using extract_point_cloud()

While this system keeps repeating these steps I can reconstruct a simple scene consisting of just a chair on the floor. After a while I can see it has reconstructed the chair well enough. Then I remove the chair while the system is running. However the system can't remove the chair from the extracted point cloud!  

This makes me wonder if the VoxelBlockGrid supports spatial carving at all. Is compute_unique_block_coordinates() just computing the block coordinates for the blocks that are on the seen surface and not for the blocks along the rays that touch the surface?   I would expect spatial carving to be part of a TSDF integration, so that the empty space between the camera and the surface is cleared.

**Comment by germanros1987** [comment]
Unfortunately, we don't have these features yet. We agree they would be a great contribution to Open3D. Let's add them to the list of feature requests.
